components/storage/blockdevice/COMPONENT_OSPIF/OSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1885 @@
+        if (ospi_status == OSPI_STATUS_OK) {
+            ospi_status = _ospi_send_general_command(0x99, OSPI_NO_ADDRESS_COMMAND, // Send reset instruction
+                                                     NULL, 0, NULL, 0);
+        }
+        status = (ospi_status == OSPI_STATUS_OK) ? OSPIF_BD_ERROR_OK : OSPIF_BD_ERROR_PARSING_FAILED;
+    } else {
+        // Soft reset either is not supported or requires direct control over data lines
+        status = OSPIF_BD_ERROR_PARSING_FAILED;
+    }
+
+    if (status == OSPIF_BD_ERROR_OK) {
+        if (false == _is_mem_ready()) {
+            tr_error("Device not ready, reset failed");
+            status = OSPIF_BD_ERROR_READY_FAILED;
+        }
+    }
+
+    return status;
+}
+
+int OSPIFBlockDevice::_sfdp_parse_4_byte_inst_table(Callback<int(bd_addr_t, void *, bd_size_t)> sfdp_reader,
+                                                    sfdp_hdr_info &sfdp_info)
+{
+    uint8_t four_byte_inst_table[SFDP_DEFAULT_4_BYTE_INST_TABLE_SIZE_BYTES]; /* Up To 2 DWORDS = 8 Bytes */
+
+    int status = sfdp_reader(sfdp_info.fbatbl.addr, four_byte_inst_table, sfdp_info.fbatbl.size);
+    if (status != OSPI_STATUS_OK) {
+        tr_error("Init - Read SFDP Four Byte Inst Table Failed");
+        return -1;
+    }
+
+    // Support 4-Byte address instructions
+    if (four_byte_inst_table[OSPIF_4_BYTE_INST_TABLE_FAST_READ_SUPPORT_BYTE] & FOURBYTE_INST_1_1_1_READ_BITMASK) {
+        _support_4_byte_inst = true;
+    }
+
+    // Detect erase types 4-Byte instructions
+    if (sfdp_info.fbatbl.size > OSPIF_4_BYTE_INST_TABLE_INST_ERASE_TYPE_1_BYTE) {
+        // Loop Erase Types 1-4
+        for (int i_ind = 0; i_ind < 4; i_ind++) {
+            sfdp_info.fbatbl.erase_type_4_byte_inst_arr[i_ind] = OSPI_NO_INST; // Default for unsupported type
+            if (four_byte_inst_table[OSPIF_4_BYTE_INST_TABLE_ERASE_SUPPORT_BYTE] & (FOURBYTE_INST_ERASE_TYPE_1_BITMASK << i_ind)) {
+                sfdp_info.fbatbl.erase_type_4_byte_inst_arr[i_ind] = four_byte_inst_table[OSPIF_4_BYTE_INST_TABLE_INST_ERASE_TYPE_1_BYTE + i_ind];
+            }
+            tr_debug("Erase Type %d - Inst: 0x%xh", (i_ind + 1), sfdp_info.fbatbl.erase_type_4_byte_inst_arr[i_ind]);
+        }
+    }
+
+    return 0;
+}
+
+int OSPIFBlockDevice::_handle_vendor_quirks()
+{
+    uint8_t vendor_device_ids[OSPI_RDID_DATA_LENGTH] = {0};
+    /* Read Manufacturer ID (1byte), and Device ID (2bytes) */
+    ospi_status_t status = _ospi_send_general_command(OSPIF_INST_RDID, OSPI_NO_ADDRESS_COMMAND,
+                                                      NULL, 0,
+                                                      (char *) vendor_device_ids, OSPI_RDID_DATA_LENGTH);
+    if (OSPI_STATUS_OK != status) {
+        tr_error("Read Vendor ID Failed");
+        return -1;
+    }
+
+    tr_debug("Vendor device ID = 0x%x 0x%x 0x%x", vendor_device_ids[0], vendor_device_ids[1], vendor_device_ids[2]);
+
+    switch (vendor_device_ids[0]) {
+        case 0xbf:
```

Comment:
Different vendor, I guess this doesn't apply anymore. But I guess this can be left if it's needed in the future.

Reply:
If this is dead code now then can it be removed please?

Reply:
It can be removed when testing with the latest MX25LM51245G